llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/Passes/MaterializeEncoding.cpp

```diff
@@ -97,6 +104,11 @@ chooseEncodingInfo(RankedTensorType tensorType) {
   case EncodingUser::MATMUL:
   case EncodingUser::BATCH_MATMUL:
     return chooseEncodingInfoForMatmul(user, role, /*tileParams=*/{8, 4, 8});
+  case EncodingUser::VECMAT:
+    return chooseEncodingInfoForMatmul(user, role, /*tileParams=*/{1, 4, 8});
+  case EncodingUser::MATVEC:
+  case EncodingUser::BATCH_MATVEC:
+    return chooseEncodingInfoForMatmul(user, role, /*tileParams=*/{8, 4, 1});
```

Comment:
This part is mostly for testing. To enable it on CPU backend, you need to add configurations to CPUMaterializeEncodingPass

https://github.com/openxla/iree/blob/3323519bad0f2b91532a8272234034920101709b/compiler/src/iree/compiler/Codegen/Common/CPU/CPUMaterializeEncodingPass.cpp#L47-L148

(I'm sorry about the structure and I will centralize the code)